odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1447,3 +1447,150 @@ def _validate_module_views(self, module):
+        """ Use the _get_faulty_views and take appropriate actions to fix the issue """
+        for view in self:
+            view_id = view.id # allow to use even if view is deleted
+            faulty_views = view._get_faulty_views(view,self.env['ir.ui.view'])
+            if faulty_views:
+                self._cr.execute('SAVEPOINT end_validation')
+                faulty_views._clean_properly(delete)
+                self._cr.execute("select count(id) from ir_ui_view where id = %s", (view_id,))
+                cnt = self.env.cr.fetchone()[0]
+                if cnt == 0 or view._is_valid():
+                    _logger.info("View %s can be fixed by disabling/deleting following views: %s" % (view_id,faulty_views))
+                    self._cr.execute('RELEASE end_validation')
+                else:
+                    self._cr.execute('ROLLBACK TO end_validation')
+                    self._cr.execute('RELEASE end_validation')
+                    _logger.warning("Issue occurs: The proposition made to fix the view %s (disable/delete %s) is not recommended" % (view,faulty_views))
+                    # should not happend
+
+    def _get_faulty_views(self,parent_view, disabled_views):
+        """ Returns the faulty views
+            To determine the faulty views, we travel the tree recursively.
+            Disable all children, then enable one by one. If there is an issue.
+            The activated node or at least one of its children are faulty.
+            Disable faulty node and continue recursion.
+            This function is never called from odoo code but allows to be called from a server action.
+            Please don't remove
+    params:
+        parent_view - ir.ui.view - Parent view used by recusion to keep the top parent view
+        disabled_views - ir.ui.view - Disabled views
+    usage:
+        view1._get_faulty_views(view1, self.env['ir.ui.view'])
+        """
+        self.ensure_one()
+        if parent_view==self:
+            if self._is_valid():
+                return disabled_views
+            self._cr.execute('SAVEPOINT test_fix_views')
+            temp_disabled_views=self._deactivate_lvl1_children()
+            if self._is_valid():
+                if self._has_active_children() or len(temp_disabled_views)>1:
+                    for child_view in (temp_disabled_views):
+                        v =child_view._get_faulty_views(self,disabled_views)
+                        for vv in v:
+                            if vv not in disabled_views: # ir.ui.view(1,2) not in ir.ui.view(1,3,4)
+                                disabled_views+=vv       # This loop avoid duplicates ir.ui.view(1,1,2,3,4)
+                else:
+                    self.active=False
+                    disabled_views+=self
+            else:
+                self.active=False
+                disabled_views+=self
+            self._cr.execute('ROLLBACK TO test_fix_views')
+            self._cr.execute('RELEASE test_fix_views')
```

Comment:
I think it is better to release explicitly. we are not sure about the context where it will be executed, prefer to no change it and not make assumption of the context